File: frontend/src/types/project.ts
export type ProjectStatus = 'active' | 'archived' | 'deleted'
export type MemberRole = 'owner' | 'editor' | 'viewer'

export interface ProjectMember {
  id: string
  user_id: string
  username: string
  full_name: string | null
  role: MemberRole
  joined_at: string
}

export interface Project {
  id: string
  name: string
  description: string | null
  status: ProjectStatus
  owner_id: string
  created_at: string
  updated_at: string
  member_count: number
  document_count: number
}

export interface ProjectDetail extends Project {
  members: ProjectMember[]
}

export interface CreateProjectRequest {
  name: string
  description?: string
}

export interface UpdateProjectRequest {
  name?: string
  description?: string
  status?: ProjectStatus
}

File: backend/app/schemas/project.py
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel


class ProjectCreate(BaseModel):
    name: str
    description: Optional[str] = None


class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None


class ProjectMemberResponse(BaseModel):
    id: str
    user_id: str
    username: str
    full_name: Optional[str]
    role: str
    joined_at: datetime

    model_config = {"from_attributes": True}


class ProjectResponse(BaseModel):
    id: str
    name: str
    description: Optional[str]
    status: str
    owner_id: str
    created_at: datetime
    updated_at: datetime
    member_count: int = 0
    document_count: int = 0

    model_config = {"from_attributes": True}


class ProjectDetailResponse(ProjectResponse):
    members: List[ProjectMemberResponse] = []


class AddMemberRequest(BaseModel):
    user_id: str
    role: str = "viewer"


class UpdateMemberRoleRequest(BaseModel):
    role: str
